src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs

```diff
@@ -490,6 +497,34 @@ private bool PerformCompilation()
             return true;
         }
 
+        private bool PerformPluginWork()
+        {
+            if (_pluginRegistration.WasAssigned ||
+                _pluginWorkNeeded.WasAssigned ||
+                _filesChanged.WasAssigned ||
+                _rebuild.WasAssigned ||
+                _restoreComplete.WasAssigned ||
+                _sourceTextChanged.WasAssigned)
```

Comment:
I'm not worried about the verbosity of the if statement. I'm not sure this is correct. No work should happen if a plugin is never registered.